test/tests.py

```diff
@@ -1,29 +1,161 @@
+            gen_list.append(tup)
+
+def unpickle(filename, expected_output):
+    with open(output_directory+filename, 'rb') as f:
+        expected_output.extend(pickle.load(f))
+
+def unjson(filename, expected_output):
+    with open(output_directory+filename) as f:
+        expected_output.append(json.load(f))
+
 
 class test_network_methods(unittest.TestCase):
 
+    def setUp(self):
+        pass
+
     def test_msg_number_graph(self):
+
         msg_number_graph = network.message_number_graph(log_data, nicks, nick_same_list)
-        expected_msg_number_graph = nx.read_pajek(output_directory + "message_number_graph.net")
-        self.assertTrue(nx.is_isomorphic(msg_number_graph, expected_msg_number_graph), msg="test passed")
+        self.assertTrue(compare_graph_outputs (msg_number_graph, "message_number_graph.gpickle"), msg="test failed")
+
+    #since test are run alphabetically append a 'a' in test name to run it first
+    #since running this test later is creating error.
+    #needs to be fixed
+    def test_a_msg_time_graph(self):
+
+        msg_time_graph = network.message_time_graph(log_data, nicks, nick_same_list)
+        self.assertTrue(compare_graph_outputs (msg_time_graph, "message_time_graph.gpickle"), msg="test failed")
+
+    def test_nick_change_graph(self):
+
+        nick_change_graph = user.nick_change_graph(log_data)
+        self.assertTrue(compare_graph_outputs (nick_change_graph, "nick_change_graph.gpickle"), msg="test failed")
+
+    def test_full_presence_graph(self):
+
+        nicks1, nick_same_list1, channels_for_user, nick_channel_dict, nicks_hash, channels_hash = nickTracker.nick_tracker(log_data, True)
+        dict_out, graph = network.channel_user_presence_graph_and_csv(nicks1, nick_same_list1, channels_for_user, nick_channel_dict, nicks_hash, channels_hash)
+
+        self.assertTrue(compare_graph_outputs (graph, "full_presence_graph.gpickle"), msg="test failed")
+
+
+class test_methods_returning_list(unittest.TestCase):
+
+    def setUp(self):
+        pass
+
+    def test_conv_len_conv_refr_time(self):
+
+        conv_len, conv_ref_time = channel.conv_len_conv_refr_time(log_data, nicks, nick_same_list)
+
+        expected_con_len = []
+        expected_conv_ref_time = []
+
+        csv_to_list (output_directory+'conv_len.csv', expected_con_len)
+        csv_to_list (output_directory+'conv_ref_time.csv', expected_conv_ref_time)
+
+        self.assertListEqual(conv_len, expected_con_len, msg="conv_len not same")
+        self.assertListEqual(conv_ref_time, expected_conv_ref_time, msg="conv_ref_time_not_same")
+
+    def test_response_time_from_channel_lib(self):
+
+        resp_time = channel.response_time(log_data, nicks, nick_same_list)
+
+        expected_resp_time = []
+
+        csv_to_list (output_directory + 'resp_time.csv', expected_resp_time)
+        self.assertListEqual(resp_time, expected_resp_time, msg="resp_time not same as expected output")
+
+    def test_message_number_bins_csv(self):
+
```

Comment:
don't add extra enters. It makes very difficult to read the code.
and use 2 enters between function definitions. Aren't you using a linter?